lib/hanami/generators/application/container/spec_helper.rb.minitest.tt

```diff
@@ -4,4 +4,6 @@ ENV['HANAMI_ENV'] ||= 'test'
 require_relative '../config/environment'
 require 'minitest/autorun'
 
+$container = Hanami::Container.new
```

Comment:
@Erol This can't be initialized here.

If we initialize `Hanami::Container` here that means that the entire framework **and all the project code** (apps, controllers, models, views) will be loaded. This is fine for _feature tests_, where we want to test the entire stack, but for single unit tests isn't the case.

In other words, when you run this unit test.

```ruby
bundle exec rspec spec/web/controllers/home/index_spec.rb
```

Hanami only loads the framework and app configurations (`Hanami::Application.preload!`). Then that unit test has a `require_relative '../../../../apps/web/controllers/home/index'` instruction to load only that object in isolation. This makes TDD possible with Hanami.

---

I understand what you're trying to do here, but instead of initialize it, we should rely on a different strategy like `Hanami::Container.preload!`. This should only look at the mounted applications and **without to initialize them**, invoke `.preload!` on them by passing the mounting path.